File: DeliveryMatch-BackEnd/src/main/java/entity/Trajet.java
package entity;

import jakarta.persistence.*;
import lombok.Data;

import java.util.Date;
import java.util.List;

@Entity
@Data
public class Trajet {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String pointDepart;

    @ElementCollection
    private List<String> etapesIntermediaires;

    @Column(nullable = false)
    private String destinationFinale;

    @Temporal(TemporalType.TIMESTAMP)
    private Date dateDepart;

    private String dimensionsMax;
    private String typeMarchandise;
    private int capaciteDisponible;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "conducteur_id", nullable = false)
    private Utilisateur conducteur;

    @OneToMany(mappedBy = "trajet", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Demande> demandesRecues;
}

File: DeliveryMatch-FrontEnd/src/app/shared/models/trajet.ts
export interface Trajet {
  id?: number;
  pointDepart: string;
  destinationFinale: string;
  etapesIntermediaires: string;
  dateDepart: string;
  capaciteDisponible: number;
  dimensionsMax: string;
  typeMarchandise: string;
  idConducteur?: number;
}
